src/crush/CrushWrapper.cc

```diff
@@ -1702,8 +1702,13 @@ int CrushWrapper::add_bucket(
   crush_bucket *b = crush_make_bucket(crush, alg, hash, type, size, items,
 				      weights);
   assert(b);
-  int r = crush_add_bucket(crush, bucketno, b, idout);
-  int pos = -1 - *idout;
+  int *pid = idout;
```

Comment:
seems the original logic will allow a null idout to be passed in,
so I am trying to minimize the changing in order to not break anything

I just did a complete search and it turns out all paths will pass in a non-null id now.
So I guess `assert(idout)` is probably okay?